src/org/whstsa/library/util/DayGenerator.java

```diff
@@ -123,12 +176,21 @@ public static BookType randomBookType() {
 
 	public static ILibrary randomLibrary() {
 		List<ILibrary> libraries = ObjectDelegate.getLibraries();
+		if (libraries.size() == 1) {
+			return libraries.get(0);
+		}
 		return libraries.get(RANDOM.nextInt(libraries.size() - 1));
 	}
 
 	public static IMember randomMember() {
 		ILibrary library = randomLibrary();
 		List<IMember> members = library.getMembers();
+		if (members.size() == 1) {
```

Comment:
This makes it so that you never get a random member; you are always getting the first member in the list.